scripts/schema-publish.sh

```diff
@@ -7,10 +7,19 @@
 schemaDir="src/schemas/validation"
 branch=$(git branch --show-current)
 
-if [[ $branch =~ ^v([0-9]+\.[0-9]+)-dev$ ]]; then
-  deploydir="./deploy/oas/${BASH_REMATCH[1]}"
-else
+
+if [ -z "$1" ]; then
+  if [[ $branch =~ ^v([0-9]+\.[0-9]+)-dev$ ]]; then
+    deploydir="./deploy/oas/${BASH_REMATCH[1]}"
+  else
+    echo "Unable to determine version from branch name; should be vMAJOR.MINOR.PATCH-dev"
+    exit 1
+  fi
+elif [ $1 = "src" ]; then
   deploydir="./deploy-preview"
+else
+  echo "Unrecognized argument"
+  exit 1
```

Comment:
This change causes the npm script `build-src` to fail.

The "old" logic of building to `deploy` in `vX.Y-dev` branches and to `deploy-preview` in all other branches seems to cover all reasonable use cases.

Why would I ever want to build to `deploy-preview` in a `vX.Y-dev` branch?

Reply:
I imagined that while developing schema versions, running `build-src` would create a predictable set of artifacts in `deploy-preview`, including the specification and its current schema.